cmd/expiration-mailer/main.go

```diff
@@ -0,0 +1,273 @@
+		}
+		err = m.mailer.SendMail(emails, msgBuf.String())
+		if err != nil {
+			m.stats.Inc("Mailer.Errors.SendingNag.SendFailure", 1, 1.0)
+			return err
+		}
+		m.stats.Inc("Mailer.Expiration.Sent", int64(len(emails)), 1.0)
+	}
+	return nil
+}
+
+func (m *mailer) updateCertStatus(serial string) error {
+	// Update CertificateStatus object
+	tx, err := m.dbMap.Begin()
+	if err != nil {
+		m.log.Err(fmt.Sprintf("Error opening transaction for certificate %s: %s", serial, err))
+		tx.Rollback()
+		return err
+	}
+
+	csObj, err := tx.Get(&core.CertificateStatus{}, serial)
+	if err != nil {
+		m.log.Err(fmt.Sprintf("Error fetching status for certificate %s: %s", serial, err))
+		tx.Rollback()
+		return err
+	}
+	certStatus := csObj.(*core.CertificateStatus)
+	certStatus.LastExpirationNagSent = time.Now()
+
+	_, err = tx.Update(certStatus)
+	if err != nil {
+		m.log.Err(fmt.Sprintf("Error updating status for certificate %s: %s", serial, err))
+		tx.Rollback()
+		return err
+	}
+
+	err = tx.Commit()
+	if err != nil {
+		m.log.Err(fmt.Sprintf("Error commiting transaction for certificate %s: %s", serial, err))
+		tx.Rollback()
+		return err
+	}
+
+	return nil
+}
+
+func (m *mailer) processCerts(certs []core.Certificate) {
+	m.log.Info(fmt.Sprintf("expiration-mailer: Found %d certificates, starting sending messages", len(certs)))
+	for _, cert := range certs {
+		regObj, err := m.dbMap.Get(&core.Registration{}, cert.RegistrationID)
+		if err != nil {
+			m.log.Err(fmt.Sprintf("Error fetching registration %d: %s", cert.RegistrationID, err))
+			m.stats.Inc("Mailer.Errors.GetRegistration", 1, 1.0)
+			continue
+		}
+		reg := regObj.(*core.Registration)
+		parsedCert, err := x509.ParseCertificate(cert.DER)
+		if err != nil {
+			m.log.Err(fmt.Sprintf("Error parsing certificate %s: %s", cert.Serial, err))
+			m.stats.Inc("Mailer.Errors.ParseCertificate", 1, 1.0)
+			continue
+		}
+		err = m.sendNags(parsedCert, reg.Contact)
+		if err != nil {
+			m.log.Err(fmt.Sprintf("Error sending nag emails: %s", err))
+			m.stats.Inc("Mailer.Errors.SendingNags", 1, 1.0)
+			continue
+		}
+		err = m.updateCertStatus(cert.Serial)
+		if err != nil {
+			m.log.Err(fmt.Sprintf("Error updating certificate status for %s: %s", cert.Serial, err))
+			m.stats.Inc("Mailer.Errors.UpdateCertificateStatus", 1, 1.0)
+			continue
+		}
+	}
+	m.log.Info("expiration-mailer: Finished sending messages")
+	return
+}
+
+func (m *mailer) findExpiringCertificates() error {
+	now := time.Now()
+	// E.g. m.NagTimes = [1, 3, 7, 14] days from expiration
+	for i, expiresIn := range m.nagTimes {
+		left := now
+		if i > 0 {
+			left = left.Add(m.nagTimes[i-1])
+		}
+		right := now.Add(expiresIn)
+
+		m.log.Info(fmt.Sprintf("expiration-mailer: Searching for certificates that expire between %s and %s", left, right))
+		var certs []core.Certificate
+		_, err := m.dbMap.Select(
+			&certs,
+			`SELECT cert.* FROM certificates AS cert
+			 JOIN certificateStatus AS cs
+			 ON cs.serial = cert.serial
```

Comment:
If we cannot use LOCATE - because we don't want to be gated on mysql in test, then we can use the slower LIKE operator:
```sql
WHERE reg.contact LIKE "%mailto%"
```